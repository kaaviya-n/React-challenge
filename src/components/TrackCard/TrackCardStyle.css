.tableContentWrapper {
    display: flex;
    align-items: center;
}
.tableValueWrapper {
    flex-grow: 1;
    padding-left: 12px;
    margin-bottom: 10px;
}
.trackName {
    line-height: 0;
    font-size: 16px;
    font-weight: 400;
    color: #ffffff;
    padding-left: 12px;
}
.trackArtist {
    font-size: 14px;
    color: #b3b3b3;
    font-weight: 400;
    margin-left: 10px;
}
.trackArtist:hover {
    border-bottom: 1px solid #b3b3b3;
}
.trackCount {
    font-size: 16px;
    font-weight: 400;
    color: #b3b3b3;
}
.trackDuration {
    font-size: 14px;
    font-weight: 400;
    color: #a7a7a7;
    margin-right: 16px;
}
.artistWrapper:last-child {
    padding-top: 10px;
}
