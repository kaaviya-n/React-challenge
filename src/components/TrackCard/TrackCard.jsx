import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import './TrackCardStyle.css';

const TrackCard = ({ track }) => {
	const { artists, name, duration_ms, track_number } = track;
	const [duration, setDuration] = useState(0);

	useEffect(() => {
		if (duration_ms) {
			var minutes = Math.floor(duration_ms / 60000);
			var seconds = ((duration_ms % 60000) / 1000).toFixed(0);
			setDuration(minutes + ':' + (seconds < 10 ? '0' : '') + seconds);
		}
	}, [duration_ms]);

	return (
		<div className='tableContentWrapper'>
			<h4 className='trackCount'>{track_number}</h4>
			<div className='tableValueWrapper'>
				<h4 className='trackName'>{name}</h4>
				<div style={{ color: '#b3b3b3', lineHeight: '0.5' }}>
					{artists.map((artist, index) => (
						<div key={index} className="artistWrapper">
							<Link
								onClick={e => e.stopPropagation()}
								to={`/artist/${artist.id}/${artist.name}`}
								className="trackArtist"
							>
								{artist.name}
							</Link>
							{index !== artists.length - 1 && ', '}
						</div>
					))}
				</div>
			</div>
			<h4 className='trackDuration'>{duration}</h4>
		</div>
	);
};

TrackCard.propTypes = {
	track: PropTypes.object
};

export default TrackCard;