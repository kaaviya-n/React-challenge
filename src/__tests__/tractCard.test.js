import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import TrackCard from '../components/TrackCard/TrackCard';

afterEach(cleanup);

const trackList = {
	'artists': [
		{
			'id': '4zCH9qm4R2DADamUHMCa6O',
			'name': 'Anirudh Ravichander',
		}
	],
	'duration_ms': 218823,
	'id': '6FQQiTpYnfc5803p84bQp1',
	'name': 'Vikram - Title Track',
	'track_number': 2
};

const TrackCardJSX = <TrackCard track={trackList} />;

it('UI testing - Track card', () => {
	const tree = renderer.create(<BrowserRouter>{TrackCardJSX}</BrowserRouter>).toJSON();
	expect(tree).toMatchSnapshot();
});

it('renders Track card component', () => {
	render(TrackCardJSX, {wrapper: BrowserRouter});
});

it('Should render without throwing an error', () => {
	expect(render(
		TrackCardJSX, {wrapper: BrowserRouter}
	)).not.toBeNull();
});
