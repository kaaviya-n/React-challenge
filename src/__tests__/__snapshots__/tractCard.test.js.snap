// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI testing - Track card 1`] = `
<div
  className="tableContentWrapper"
>
  <h4
    className="trackCount"
  >
    2
  </h4>
  <div
    className="tableValueWrapper"
  >
    <h4
      className="trackName"
    >
      Vikram - Title Track
    </h4>
    <div
      style={
        Object {
          "color": "#b3b3b3",
          "lineHeight": "0.5",
        }
      }
    >
      <div
        className="artistWrapper"
      >
        <a
          className="trackArtist"
          href="/artist/4zCH9qm4R2DADamUHMCa6O/Anirudh Ravichander"
          onClick={[Function]}
        >
          Anirudh Ravichander
        </a>
      </div>
    </div>
  </div>
  <h4
    className="trackDuration"
  >
    0
  </h4>
</div>
`;
